app/src/main/java/pm/gnosis/heimdall/ui/safe/helpers/RecoverSafeOwnersHelper.kt

```diff
@@ -184,17 +194,23 @@ class DefaultRecoverSafeOwnersHelper @Inject constructor(
         val remainingNewAddresses = addressesToAdd.toMutableList()
         for (i in safeInfo.owners.size - 1 downTo 0) {
             val safeOwner = safeInfo.owners[i]
-            if (remainingNewAddresses.isEmpty()) break
-            if (!addressesToKeep.contains(safeOwner)) {
-                val newOwner = remainingNewAddresses.removeAt(remainingNewAddresses.size - 1)
-                if (safeInfo.owners.contains(newOwner)) continue
-                val pointerAddress = nullOnThrow { safeInfo.owners[i - 1] } ?: SENTINEL
-                payloads += GnosisSafePersonalEdition.SwapOwner.encode(
-                    prevOwner = pointerAddress,
-                    oldOwner = safeOwner,
-                    newOwner = newOwner
-                )
+            if (addressesToKeep.contains(safeOwner)) continue
+            if (addressesToAdd.contains(safeOwner)) continue
+
+            var newOwner: Solidity.Address? = null
```

Comment:
We should add comments to this (maybe add a issue for that, as it is not super important right now, but maybe when doing the refactorings it would be nice)